pyccel/codegen/printing/ccode.py

```diff
@@ -998,6 +1004,41 @@ def find_in_ndarray_type_registry(self, dtype, prec):
                     severity='fatal')
 
     def get_declare_type(self, expr):
+        """
+        Get the string which describes the type in a declaration.
+
+        This function returns the code which describes the type
+        of the `expr` object such that the declaration can be written as:
+        `f"{self.get_declare_type(expr)} {expr.name}"`
+        The function takes care of reporting errors for unknown types and
+        importing any necessary additional imports (e.g. stdint/ndarrays).
+
+        Parameters
+        ----------
+        expr : Variable
+            The variable whose type should be described.
+
+        Returns
+        -------
+        str
+            The code describing the type.
+
+        Raises
+        ------
+        PyccelCodegenError
+            If the type is not supported in the C code or the rank is too large.
+
+        Examples
+        --------
+        >>> v = Variable('int', 'x')
+        >>> self.get_declare_type(v)
+        'int64_t '
+
+        For an object accessed via a pointer:
+        >>> v = Variable('int', 'x', is_optional=True, rank=1)
+        >>> self.get_declare_type(v)
+        't_ndarray* '
```

Comment:
Does the returned string end with an empty space? According to the docstring that should not be necessary.

Reply:
Yes it does. Personally I think it is better for the context to decide whether the space is necessary but that is not how it is currently coded. I have changed this : 4e6aacf3